webui/react/src/components/UPlot/UPlotChart.tsx

```diff
@@ -63,145 +67,98 @@ const shouldUpdate = (
   const scalesZoomData = useRef<Record<string, ScaleZoomData>>({});
-
-  const getAugmentedOptions = (options: Partial<uPlot.Options> | undefined)
-  : uPlot.Options => uPlot.assign(
-    {
-      hooks: {
-        setScale: [ (uPlot: uPlot, scaleKey: string) => {
-          if (![ 'x', 'y' ].includes(scaleKey)) return;
-
-          const currentMax: number|undefined =
-            uPlot.posToVal(scaleKey === 'x' ? uPlot.bbox.width : 0, scaleKey);
-          const currentMin: number|undefined =
-            uPlot.posToVal(scaleKey === 'x' ? 0 : uPlot.bbox.height, scaleKey);
-          let max: number|undefined = scalesZoomData.current[scaleKey]?.max;
-          let min: number|undefined = scalesZoomData.current[scaleKey]?.min;
-
-          if (max == null || currentMax > max) max = currentMax;
-          if (min == null || currentMin < min) min = currentMin;
-
-          scalesZoomData.current[scaleKey] = {
-            isZoomed: currentMax < max || currentMin > min,
-            max,
-            min,
-          };
-        } ],
-      },
-      width: chartDivRef.current?.offsetWidth,
+  const dataRef = useRef<uPlot.AlignedData>(EMPTY_DATA);
+  const [ isEmpty, setIsEmpty ] = useState(true);
+
+  const fullOptions = useMemo(() => uPlot.assign({
+    hooks: {
+      setScale: [ (uPlot: uPlot, scaleKey: string) => {
+        if (![ 'x', 'y' ].includes(scaleKey)) return;
+
+        const boundWidth = scaleKey === 'x' ? uPlot.bbox.width : 0;
+        const boundHeight = scaleKey === 'x' ? 0 : uPlot.bbox.height;
+        const currentMax: number | undefined = uPlot.posToVal(boundWidth, scaleKey);
+        const currentMin: number | undefined = uPlot.posToVal(boundHeight, scaleKey);
+        let max: number | undefined = scalesZoomData.current[scaleKey]?.max;
+        let min: number | undefined = scalesZoomData.current[scaleKey]?.min;
+
+        if (max == null || currentMax > max) max = currentMax;
+        if (min == null || currentMin < min) min = currentMin;
+
+        scalesZoomData.current[scaleKey] = {
+          isZoomed: currentMax < max || currentMin > min,
+          max,
+          min,
+        };
+      } ],
     },
-    options || {},
-  ) as uPlot.Options;
+  }, options ?? {}) as uPlot.Options, [ options ]);
 
-  const optionsRef = useRef<uPlot.Options>(getAugmentedOptions(options));
+  const prevFullOptions = usePrevious(fullOptions, {} as uPlot.Options);
 
-  const [ hasData, normalizedData ] = useMemo(
-    () => getNormalizedData(data, optionsRef.current?.mode)
-    , [ data ],
-  );
-
-  /*
+  /**
    * Chart mount and dismount.
    */
   useEffect(() => {
     if (!chartDivRef.current) return;
-    scalesZoomData.current = {};
-    const data = [ [], [ [] ] ] as unknown as uPlot.AlignedData;
-    if (!chartRef?.current) {
-      chartRef.current = new uPlot(optionsRef.current, data, chartDivRef.current);
-      setChartIsMounted(true);
-    }
+    if (!shouldRecreate(chartRef.current, fullOptions, prevFullOptions)) return;
```

Comment:
this is conditional and the destroy is not so if this returns false the charts will be destroyed. 

`if (!chartRef?.current) {` is conditional too but only on the chart existing in the first place